crates/interpreter/src/instructions/macros.rs

```diff
@@ -30,19 +28,19 @@ macro_rules! gas {
 }
 
 macro_rules! refund {
-    ($interp:expr, $gas:expr) => {{
+    ($interp:expr, $gas:expr) => {
         if crate::USE_GAS {
-            $interp.gas.gas_refund($gas);
+            $interp.gas.record_refund($gas);
         }
-    }};
+    };
 }
 
 macro_rules! gas_or_fail {
     ($interp:expr, $gas:expr) => {
         if crate::USE_GAS {
             match $gas {
-                Some(gas_used) => gas!($interp, gas_used),
-                None => {
+                Some(gas_used) if $interp.gas.record_cost(gas_used) => {}
+                _ => {
```

Comment:
nit any reason to not use the macro

Reply:
Would duplicate the assign/return on both branches, doesn't really matter since compiler probably optimizes that anyway

Reply:
Can you revert this one, I slightly vary on the fn call side-effect inside if, and second reason is that `gas!` is used everywhere so it converges to one place

Reply:
Understandable

Reply:
fac35ad7331a39d91eb353a9e9d5366127730cda